### PR TITLE
Github actions enhancements (kind->GKE, staging cluster)

### LINKED ISSUE #424: Pull request CI is failing with KIND "unable to get nodes" error
https://github.com/GoogleCloudPlatform/microservices-demo/pull/423/checks?check_run_id=1344754071

### PR DESCRIPTION
**Pull request by askmeegs**
This is a CI overhaul, the purpose is to standardize this app's testing to what we already have in Bank of Anthos, in preparation for onboarding new contributors soon. (Who will be working on both apps at once.)

Fixes #424 and adds the following enhancements: 

- [x] a new administrative GCP project dedicated to CI/CD for this repo. 
- [x] in that project, 2 GKE clusters. one for running CI deploy tests (PR cluster), one for hosting onlineboutique.dev (master cluster)
- [x] set up onlineboutique.dev to point to a global static IP (which points to the frontend ingress for the master cluster running the v0.2.0 images). Also set up SSL certs for the domain and provided manifests in case we ever have to do it again.  
- [x] in that project, 2 new GCE custom github actions runners. removed the old runners. 
- [x] an updated install dependencies script with the dotnet CLI
- [x] updated github actions workflows for ci-pr and ci-master. Only difference is that the ci-master build doesn't stage the build or trigger a PR comment, because there's no PR involved.  Both use **GKE instead of kind** for the deploy tests. The benefit of this is more realistically testing the target K8s this demo was built for, being able to stage PRs for people to see their code changes in the UI, and not having to spin up a new cluster inside GCE repeatedly. The PR cluster is persistent and has autoscaling on to support multiple active PRs and therefore multiple app deployments in different namespaces. 
- [x] added a push and deploy latest workflow to push the `:latest` image tag off master, to the `onlineboutique-ci` GCR. **Note** - right now, because these runners don't currently have push access to the `google-samples` project. 

## remaining work 

- get the dotnet tests working in CI for cartservice, currently seeing relative path errors, possible dotnet core version mismatch #428
- fix the productcatalog unit tests which currently fail on go 1.15  - #426 
- extend the unit tests which right now are minimal #427

### CODE REVIEW

**Comment by ahmetb** [comment]
This looks superb. I'm not sure how/where the domain name and the new GCP project is managed, if you haven't hashed that out yet, I recommend creating an internal doc and making it findable on internal search engine.

**Comment by askmeegs** [comment]
Hey Ahmet, yep planning to add a doc on the onlineboutique CI/CD project, and contact info for the domain (which is currently with my Google account.)

File: .github/workflows/README.md [hunk 1]
```diff
@@ -1,75 +1,61 @@
...
-  - ensures all pods reach ready state
-  - ensures HTTP request to frontend returns HTTP status 200
-- deploys manifests from /releases
-  - ensures all pods reach ready state
-  - ensures HTTP request to frontend returns HTTP status 200
+This page describes the CI/CD workflows for the Online Boutique app, which run in [Github Actions](https://github.com/GoogleCloudPlatform/microservices-demo/actions). 
+
+## Infrastructure 
+
+The CI/CD pipelines for Online Boutique run in Github Actions, using a pool of two [self-hosted runners]((https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-self-hosted-runners)). These runners are GCE instances (virtual machines) that, for every open Pull Request in the repo, run the code test pipeline, deploy test pipeline, and (on master) deploy the latest version of the app to [onlineboutique.dev](https://onlineboutique.dev)
```
Comment (djmailhot) [inline .github/workflows/README.md:7]: So do you foresee keeping these self-hosted runners operating for as long as this repo exists?

Do you know if the runners are persistent on GCE VMs?  (When GCE decides to migrate the runner VM, will the self-hosted runner process be uninterrupted and continue to serve requests?)

**Review by djmailhot — APPROVED** [review]
Looks fantastic.

One observation.  Other than that, LGTM

**Pull request merged**